packages/hawtio/src/plugins/shared/workspace.ts

```diff
@@ -46,6 +54,68 @@ class Workspace {
     })
   }
 
+  private maybeUpdatePlugins(response: IResponse): void {
+    if (this.pluginUpdateCounter === null) {
+      this.pluginUpdateCounter = response.value as number | null
+      return
+    }
+    if (this.pluginUpdateCounter !== response.value) {
+      if (parseBoolean(localStorage['autoRefresh'])) {
+        window.location.reload()
+      }
+    }
+  }
+
+  private maybeReloadTree(response: IResponse): void {
+    const counter = response.value
+    if (this.treeWatcherCounter === null) {
+      this.treeWatcherCounter = counter as number | null
+      return
+    }
+    if (this.treeWatcherCounter !== counter) {
+      this.treeWatcherCounter = counter as number | null
+      this.refreshTree()
+    }
+  }
+
+  async maybeMonitorPlugins() {
```

Comment:
This method name doesn't tell what it really does. What about splitting the method into two: `maybeMonitorPlugins()` and `maybeMonitorTree()`?